packages/atomic/src/components/common/query-summary/query-summary-common.tsx

```diff
@@ -0,0 +1,124 @@
+import {FunctionalComponent, h} from '@stencil/core';
+import {AnyBindings} from '../interface/bindings';
+import escape from 'escape-html';
+
+interface QuerySummaryCommonProps {
+  bindings: AnyBindings;
+  enableDuration: boolean;
+  querySummaryState: {
+    hasDuration: boolean;
+    durationInSeconds: number;
+    lastResult: number;
+    firstResult: number;
+    total: number;
+    query: string;
+    isLoading: boolean;
+    hasQuery: boolean;
+    hasError: boolean;
+    firstSearchExecuted: boolean;
+    hasResults: boolean;
+  };
+  setAriaLive: (message: string) => void;
+}
+
+export const QuerySummaryCommon: FunctionalComponent<
+  QuerySummaryCommonProps
+> = (props) => {
+  const getDurationInSeconds = (count: number) =>
+    props.bindings.i18n.t('in-seconds', {count});
+
+  const renderDuration = () => {
+    const shouldShow =
+      props.enableDuration && props.querySummaryState.hasDuration;
+    return `<span class="${
+      shouldShow ? '' : 'hidden'
+    }" part="duration"> ${getDurationInSeconds(
+      props.querySummaryState.durationInSeconds
+    )}</span>`;
+  };
+
+  const wrapHighlight = (content: string) => {
+    return `<span class="font-bold" part="highlight">${content}</span>`;
+  };
+
+  const wrapQuery = (content: string) => {
+    return `<span class="font-bold" part="highlight query">${content}</span>`;
+  };
```

Comment:
If you use the LocalizedString util, these could become scoped functional components. E.g.:
```tsx
const Highlighted = ({isQuery, children}: PropsWithChildren<{ isQuery: boolean }>) =>
  <span class="font-bold" part={isQuery ? 'highlight query' : 'highlight'}>{children}</span>

...

return <LocalizedString
  bindings={props.bindings}
  key="..."
  params={{
    someValue: <Highlighted isQuery={false}>{props.querySummaryState.something}</Highlighted>
  }}
/>
```